sharedresource/v1alpha1/types_share.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright The OpenShift authors.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Share is the Schema for the shares API
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +kubebuilder:subresource:status
+// +openshift:compatibility-gen:level=4
+type Share struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec   ShareSpec   `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+	Status ShareStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ShareList contains a list of Share
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +kubebuilder:subresource:status
+// +openshift:compatibility-gen:level=4
+type ShareList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	Items           []Share `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// ShareSpec defines the desired state of Share
+type ShareSpec struct {
+	// BackingResource captures the ConfigMap or Secret that is shared.
+	// +required
+	BackingResource `json:"backingResource" protobuf:"bytes,1,opt,name=backingResource"`
+
+	// Description is a user readable explanation of what the backing resource
+	// provides.
+	// +optional
+	Description string `json:"description,omitempty" protobuf:"bytes,2,opt,name=description"`
+}
+
+// ShareStatus defines the observed state of Share
+type ShareStatus struct {
+	// Conditions are the set of k8s Condition instances provided by the associated controller for Shares.
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty" protobuf:"bytes,1,rep,name=conditions"`
+}
+
+type BackingResource struct {
+	// Kind is a string value representing the REST resource this object represents.
+	// Currently only Secret and ConfigMap are accepted.
+	// +required
+	Kind string `json:"kind" protobuf:"bytes,1,opt,name=kind"`
+
+	// APIVersion defines the versioned schema of this representation of an object.
+	// +optional
+	APIVersion string `json:"apiVersion,omitempty" protobuf:"bytes,2,opt,name=apiVersion"`
```

Comment:
You should have a separate group field.

You only need version if the serialization format of the content is required.  It's not obvious to me whether this is the case or not.

Reply:
updated in next push